kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ProxyChannelStateMachine.java

```diff
@@ -102,20 +106,58 @@
  *     Thus when the proxy is notified that a peer is applying back pressure it results in action on the channel with the opposite peer.
  * </p>
  */
+@SuppressWarnings("java:S1133")
 public class ProxyChannelStateMachine {
     private static final String DUPLICATE_INITIATE_CONNECT_ERROR = "NetFilter called NetFilterContext.initiateConnect() more than once";
     private static final Logger LOGGER = getLogger(ProxyChannelStateMachine.class);
+    /**
+     * @deprecated use `clientToProxyConnectionCounter` instead
+     */
+    @Deprecated(since = "0.13.0", forRemoval = true)
     private final Counter downstreamConnectionsCounter;
+    /**
+     * @deprecated use `proxyToServerConnectionCounter` instead
+     */
+    @Deprecated(since = "0.13.0", forRemoval = true)
     private final Counter upstreamConnectionsCounter;
+    /**
+     * @deprecated use `clientToProxyErrorCounter` instead
+     */
+    @Deprecated(since = "0.13.0", forRemoval = true)
     private final Counter downstreamErrorCounter;
+    /**
+     * @deprecated use `proxyToServerErrorCounter` instead
+     */
+    @Deprecated(since = "0.13.0", forRemoval = true)
     private final Counter upstreamErrorCounter;
+    /**
+     * @deprecated use `proxyToServerConnectionCounter` instead
+     */
+    @Deprecated(since = "0.13.0", forRemoval = true)
     private final Counter connectionAttemptsCounter;
+    /**
+     * @deprecated use `proxyToServerConnectionCounter` instead
+     */
+    @Deprecated(since = "0.13.0", forRemoval = true)
     private final Counter upstreamConnectionFailureCounter;
+    private final Counter clientToProxyErrorCounter;
+    private final Counter clientToProxyConnectionCounter;
+    private final Counter proxyToServerConnectionCounter;
+    private final Counter proxyToServerErrorCounter;
 
-    public ProxyChannelStateMachine(String clusterName) {
+    @SuppressWarnings("java:S5738")
+    public ProxyChannelStateMachine(String clusterName, @Nullable Integer nodeId) {
         List<Tag> tags = Metrics.tags(Metrics.VIRTUAL_CLUSTER_TAG, clusterName);
         downstreamConnectionsCounter = taggedCounter(KROXYLICIOUS_DOWNSTREAM_CONNECTIONS, tags);
+        clientToProxyConnectionCounter = KROXYLICIOUS_CLIENT_TO_PROXY_CONNECTION_TOTAL_METER_PROVIDER
+                .create(clusterName, nodeId).withTags();
```

Comment:
I dunno I still wonder if a three arg option would be better
```
clientToProxyConnectionCounter = meterProvider.create(meterName, clusterName, nodeId).withTags("specific tag 1"
```

Reply:
What would set the description?

Reply:
Nothing says the create has to actually build the meter right? `Metrics` could map the name to a description internally and ensure a standard set of tags. 

This is good enough for now.